### PR TITLE
feat: implement Detail blog page

### PR DESCRIPTION
**Pull request by milo219**

### CODE REVIEW

File: src/components/DetailBlog/index.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+              </li>
+              <li>
+                <a href="/">dinner</a>
+              </li>
+              <li>
+                <a href="/">lunch</a>
+              </li>
+            </ul>
+          </div>
+          {/* <include src="components/like-button.html"></include> */}
```
Comment (demons1502) [inline src/components/DetailBlog/index.tsx:68]: remove

File: src/components/SocialShare/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+function SocialShare() {
+  return (
+    <div className="social-share mb-4">
+      <div className="widget-social">
+        <p>Share: </p>
+        <a title="Twitter" href="/">
```
Comment (demons1502) [inline src/components/SocialShare/index.tsx:8]: define a constant and map it

File: src/components/DetailBlog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+interface Props {
+  imgDetailBlog: string;
+  tittleDetailBlog: string;
+  info: string;
+  day: string;
+  numberComment: string;
```
Comment (demons1502) [inline src/components/DetailBlog/index.tsx:8]: type of `numberComment` is string ?

File: src/components/DetailBlog/index.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  content2,
+  linkInfo,
+}: Props) {
+  return (
+    <div className="blog-detail blog-details-wrapper">
+      <div className="blog-details-content">
+        <div className="blog-details-content__image">
+          <img
+            src={imgDetailBlog}
+            alt=""
```
Comment (demons1502) [inline src/components/DetailBlog/index.tsx:31]: Add alt

File: src/components/DetailBlog/index.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        <p>{content}</p>
+
+        <blockquote> {blockquote}</blockquote>
+
+        <p>{content2}</p>
+
+        <div className="blog-details-content__tagsocial mb-n4">
+          <div className="blog-details-content__tagsocial-tag mb-4">
+            <ul>
+              <li>
```
Comment (demons1502) [inline src/components/DetailBlog/index.tsx:57]: define constant and map its

File: src/components/DetailComment/index.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  comment: string;
+}
+function DetailComment({ imgAvtComment, info, day, comment }: Props) {
+  return (
+    <div className="blog-details-comment mt-5">
+      <figure className="blog-details-comment thumb">
+        <img
+          className="thumb__fit-image img-fluid rounded-circle mr-3 mb-3"
+          src={imgAvtComment}
+          alt=""
```
Comment (demons1502) [inline src/components/DetailComment/index.tsx:16]: Same

File: src/App.tsx [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
-import Blogpage from './pages/Blogpage';
+import BlogPage from './pages/BlogPage';
```
Comment (baonguyen26) [inline src/App.tsx:3]: Check order import

File: src/components/DetailBlog/index.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import dataMock from '../../dataMock';
+import LikeButton from '../LikeButton';
+import TagSocial from '../TagSocial';
+
+interface Props {
+  imgDetailBlog: string;
```
Comment (baonguyen26) [inline src/components/DetailBlog/index.tsx:7]: `image` instead

File: src/components/DetailBlog/index.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  tittleDetailBlog: string;
```
Comment (baonguyen26) [inline src/components/DetailBlog/index.tsx:8]: `title` instead

File: src/components/DetailBlog/index.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+  info: string;
+  day: string;
+  numberComment: number;
+  content: string;
+  blockquote: string;
+  content2: string;
```
Comment (baonguyen26) [inline src/components/DetailBlog/index.tsx:14]: second?

File: src/components/DetailComment/index.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+interface Props {
+  imgAvtComment: string;
```
Comment (baonguyen26) [inline src/components/DetailComment/index.tsx:4]: naming?

File: src/components/LikeButton/index.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import React, { useState } from 'react';
+import { AiTwotoneLike } from 'react-icons/ai';
```
Comment (baonguyen26) [inline src/components/LikeButton/index.tsx:2]: Do you have another way to handle it?

File: src/components/LikeButton/index.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+function LikeButtonCompoent() {
+  const [like, setLike] = useState(0);
```
Comment (baonguyen26) [inline src/components/LikeButton/index.tsx:5]: type of state? 
`const [like, setLike] = useState<number>(0);`

File: src/components/LikeButton/index.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+  const [isLike, setIsLike] = useState(false);
```
Comment (baonguyen26) [inline src/components/LikeButton/index.tsx:6]: same here

File: src/components/LikeButton/index.tsx [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+  const onLikeButtonClick = () => {
+    setLike(like + (isLike ? -1 : 1));
+    setIsLike(!isLike);
```
Comment (baonguyen26) [inline src/components/LikeButton/index.tsx:9]: `setIsLike((like) => !like);`

File: src/components/LikeButton/index.tsx [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+      type="button"
+      className={`like-button ${isLike ? 'liked' : ''}`}
+      onClick={onLikeButtonClick}
+    >
+      Like <AiTwotoneLike className="ai-twotone-like" /> {like}
+    </button>
+  );
+}
+
+export default function App() {
```
Comment (baonguyen26) [inline src/components/LikeButton/index.tsx:23]: ???

File: src/components/LikeButton/index.tsx [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+  const [isLike, setIsLike] = useState(false);
+  const onLikeButtonClick = () => {
+    setLike(like + (isLike ? -1 : 1));
+    setIsLike(!isLike);
+  };
+
+  return (
+    <button
+      type="button"
+      className={`like-button ${isLike ? 'liked' : ''}`}
```
Comment (baonguyen26) [inline src/components/LikeButton/index.tsx:15]: define constant
eg: 
`const likeButtonClassName = `like-button ${isLike ? 'liked' : ''}``

**Review by baonguyen26 — CHANGES_REQUESTED** [review]

File: src/components/DetailBlog/index.tsx [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+interface Props {
+  img: string;
+  title: string;
+  info: string;
+  day: string;
+  numberComment: number;
+  firtContent: string;
+  blockquote: string;
+  SecondContent: string;
```
Comment (baonguyen26) [inline src/components/DetailBlog/index.tsx:16]: lowercase

File: src/components/LikeButton/index.tsx [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+import { AiTwotoneLike } from 'react-icons/ai';
+
+export default function LikeButtonComponent() {
+  const [like, setLike] = useState<number>(0);
+  const [isLike, setIsLike] = useState<boolean>(false);
+  const onLikeButtonClick = () => {
+    setLike(like + (isLike ? -1 : 1));
+    setIsLike((islike) => !islike);
+  };
+  const likeButtonClassName = `like-button ${isLike ? 'liked' : ''}`;
```
Comment (baonguyen26) [inline src/components/LikeButton/index.tsx:12]: `isLike && 'liked'`

**Review by baonguyen26 — APPROVED** [review]

**Pull request merged**